loki/backend/cgen.py

```diff
@@ -183,7 +190,10 @@ def visit_Subroutine(self, o, **kwargs):
 
         # Fill the body
         body += [self.visit(o.body, **kwargs)]
-        body += [self.format_line('return 0;')]
+
+        # if something to be returned, add 'return <var>' statement
+        if o.result_name is not None:
```

Comment:
Do we want to double-check wether `o.is_function`?

Reply:
Good point, since we can now rely on `o.result_name` to always have a sensible value for functions, we don't have to check for result_name but can rather check for is_function